radix-engine/src/engine/kernel.rs

```diff
@@ -978,20 +889,131 @@ where
+            static_refs.insert(GlobalAddress::Component(SYS_SYSTEM_COMPONENT));
+
+            // Make refs visible
+            let mut global_references = input.global_references();
+            global_references.extend(static_refs.clone());
+
+            // TODO: This can be refactored out once any type in sbor is implemented
+            let maybe_txn: Result<TransactionProcessorRunInput, DecodeError> =
+                scrypto_decode(&input.raw);
+            if let Ok(input) = maybe_txn {
+                for instruction in &input.instructions {
+                    match instruction {
+                        Instruction::CallFunction { args, .. }
+                        | Instruction::CallMethod { args, .. } => {
+                            let scrypto_value =
+                                ScryptoValue::from_slice(&args).expect("Invalid CALL arguments");
+                            global_references.extend(scrypto_value.global_references());
+                        }
+                        _ => {}
+                    }
+                }
+            }
+
+            // Check for existence
+            for global_address in global_references {
+                let node_id = RENodeId::Global(global_address);
+                let substate_id = SubstateId::Global(global_address);
+                let node_pointer = RENodePointer::Store(node_id);
+
+                // TODO: static check here is to support the current genesis transaction which
+                // TODO: requires references to dynamically created resources. Can remove
+                // TODO: when this is resolved.
+                if !static_refs.contains(&global_address) {
+                    node_pointer
+                        .acquire_lock(substate_id.clone(), false, false, &mut self.track)
+                        .map_err(|e| match e {
+                            KernelError::SubstateError(TrackError::NotFound(..)) => {
+                                RuntimeError::KernelError(KernelError::GlobalAddressNotFound(
+                                    global_address,
+                                ))
+                            }
+                            _ => RuntimeError::KernelError(e),
+                        })?;
+                    node_pointer
+                        .release_lock(substate_id, false, &mut self.track)
+                        .map_err(RuntimeError::KernelError)?;
+                }
+
+                Self::current_frame_mut(&mut self.call_frames)
+                    .node_refs
+                    .insert(node_id, node_pointer);
+                next_node_refs.insert(node_id, node_pointer);
+            }
+        } else {
+            // Check that global references are owned by this call frame
+            let mut global_references = input.global_references();
+            global_references.insert(GlobalAddress::Resource(RADIX_TOKEN));
+            global_references.insert(GlobalAddress::Component(SYS_SYSTEM_COMPONENT));
```

Comment:
Why do we have to make these two global addresses special?

Reply:
A hack for now, required because some of our Scrypto directly addresses these addresses in code (rather than stored in Component or passed in function)

Reply:
Okay.